dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/LayerBase.java

```diff
@@ -17,18 +17,25 @@
 
 import edu.snu.dolphin.dnn.blas.Matrix;
 
+import static edu.snu.dolphin.dnn.util.NeuralNetworkUtils.shapeFromString;
+
 /**
  * Abstract class for the layer of a neural network.
  */
 public abstract class LayerBase {
 
   private final int index;
-  private final int numOutput;
   private LayerParameter layerParameter;
+  private final int[] inputShape;
+
+  protected LayerBase(final int index, final int[] inputShape) {
```

Comment:
This constructor is used in constructors in all layers. `super(index, inputShape)`.